eng/common/TestResources/New-TestResources.ps1

```diff
@@ -156,7 +156,7 @@ if ($ProvisionerApplicationId) {
     }
 
     $provisionerAccount = Retry {
-        Connect-AzAccount -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal -Environment $Environment @subscriptionArgs
+        Connect-AzAccount -Force -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal -Environment $Environment @subscriptionArgs
```

Comment:
Less of a nit: why unconditionally? Normally, we should tie this with our own `-Force`, which you can do like so:

```powershell
Connect-AzAccount ... -Force:$Force
```

This would apply below as well. By default, cmdlets shouldn't forcibly overwrite/destroy anything, which you're doing here and below. Is there a good reason we should always pass `-Force`?